### PR TITLE
Rampslide sparks (take 2)

### PR DESCRIPTION
**Pull request by squeek502**
Updated duplicate of #48, with some spark improvements and networking

Things that could/should be done:
- [x] Emit something other than sparks (dust) when sliding on grass/dirt/etc. This would probably entail moving all the tracing client-side and removing the networked var
- [x] Add client cvars to toggle rampslide sparks or control their size/etc

### CODE REVIEW

File: game_shared/ff/ff_gamemovement.cpp [hunk 1]
```diff
@@ -907,6 +908,42 @@ bool CFFGameMovement::CanAccelerate()
 	return true;
 }
 
+bool CFFGameMovement::IsRampSliding( CFFPlayer *pPlayer )
+{
+	if (pPlayer->GetGroundEntity() == NULL)  // Rampsliding occurs when normal ground detection fails
+	{
+		// Take the lateral velocity
+		Vector vecVelocity = mv->m_vecVelocity * Vector(1.0f, 1.0f, 0.0f);
+		float flHorizontalSpeed = vecVelocity.Length();
```
Comment (AfterShockFF) [inline game_shared/ff/ff_gamemovement.cpp:917]: Minor - its possible to reduce the nesting here quite a lot by inverting the ifs and returning false immediately, e.g.

```
if (pPlayer->GetGroundEntity() != NULL) // Rampsliding occurs when normal ground detection fails
    return false;

// Take the lateral velocity
Vector vecVelocity = mv->m_vecVelocity * Vector(1.0f, 1.0f, 0.0f);
float flHorizontalSpeed = vecVelocity.Length();
if (flHorizontalSpeed <= SV_TRIMPTRIGGERSPEED)
    return false;

// etc
```

**Comment by AfterShockFF** [comment]
Other than a minor comment this is all good :+1:

**Comment by BullyFF** [comment]
https://soundcloud.com/tudberry/sets/ffsoundstuff

**Comment by squeek502** [comment]
Nice, thanks. Can you make them downloadable (in the Permissions tab of each track)?

**Comment by BullyFF** [comment]
Oops! Done :)

**Comment by AfterShock2** [comment]
@squeek502 is this ready to go in?

**Comment by squeek502** [comment]
Yeah, adding it now would probably be fine. We can add sounds afterwards.

**Pull request merged**